src/ClientHandle.cpp

```diff
@@ -318,84 +322,100 @@ void cClientHandle::Kick(const AString & a_Reason)
 
 void cClientHandle::Authenticate(const AString & a_Name, const AString & a_UUID, const Json::Value & a_Properties)
 {
-	if (m_State != csAuthenticating)
+	cWorld * World;
 	{
```

Comment:
It limits the `lock(m_CSState)`  on the next line only to lock within this scope, rather than till the function end. Specifically, I want to avoid calling plugins (line 433) with this lock held as much as possible.